src/processes/clustered.jl

```diff
@@ -0,0 +1,31 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+   PoissonClusterProcess(p, o, dₒ)
```

Comment:
Is this clustering mechanism specific to Poisson? Maybe we should just name it ClusteredProcess?

Also, if this is a general mechanism that applies to all point processes, we can introduce a new operation `hclust(parent, offspring, parentfun)` that returns the `ClusteredProcess` and does some specializations sometimes depending on the constituent processes. What do you think?

Reply:
Section 5.5.1 of Spatial Point Patterns calls this process a "Matérn cluster process".